### ISSUE TITLE
Microsoft.Insights

### ISSUE THREAD

**Issue by JamesDLD** [Azure/arm-template-whatif#89]
## Describe the noise

**Resource type**
Microsoft.Insights

**apiVersion** 
2018-05-01-preview

**Client**
PowerShell

**Relevant ARM Template code (we only need the resource object for the above `resourceType` and `apiVersion`, but if it's easier you can include the entire template**
Complete template sample is available [here](https://github.com/Azure/azure-quickstart-templates/blob/master/101-function-premium-vnet-integration/azuredeploy.json)

```
        {
            "type": "Microsoft.Insights/components",
            "apiVersion": "2018-05-01-preview",
            "location": "[parameters('location')]",
            "name": "[variables('applicationInsightsName')]",
            "tags": "[union(variables('additional_tags'),resourceGroup().tags)]",
            "kind": "web",
            "properties": {
                "Application_Type": "web",
                "ApplicationId": "[variables('applicationInsightsName')]"
            }
        }
```

**Expected response**
I expected no noise since the template has not been modified since the resources were deployed

**Current (noisy) response**
```
  ~ Microsoft.Insights/components/whp-corp-dev-poly-fnapp1 [2018-05-01-preview]
    + properties.Flow_Type:      "Bluefield"
    + properties.Request_Source: "rest"
```

**Additional context**
Thank you for this feature, I was a fan of Terraform so I really appreciate the work done here.

**Comment by alex-frankel** [comment]
Can you double check that the template is correct? Both `properties.Flow_Type` and `properties.Request_Source` are not in the template that you linked to

**Comment by JamesDLD** [comment]
Yes I confirm those are not in the template but the output appear like this.

**Comment by alex-frankel** [comment]
Do you have any `modify` or `append` policies configured in this subscription that may be augmenting the resource properties?

**Comment by JamesDLD** [comment]
I don’t think so, to be sure I will deploy de full arm template in another subscription /tenant, I don’t have access to my laptop, will do that on Monday.

**Comment by JamesDLD** [comment]
Yes I confirm, you can view it by launching the following commands.

```
# First creation
New-AzResourceGroupDeployment -Name "whatif89" -ResourceGroupName $RgName `
  -TemplateUri https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-function-premium-vnet-integration/azuredeploy.json `
  -appName "whatif89" `
  -vnetName "debugvnet1" `
  -subnetName "debugsnet1" `
  -ErrorAction Stop


# What If
New-AzResourceGroupDeployment -Name "whatif89" -ResourceGroupName $RgName `
  -TemplateUri https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-function-premium-vnet-integration/azuredeploy.json `
  -appName "whatif89" `
  -vnetName "debugvnet1" `
  -subnetName "debugsnet1" `
  -WhatIf  -ErrorAction Stop
```

<img width="722" alt="image" src="https://user-images.githubusercontent.com/18073917/85257492-3eb5ff00-b466-11ea-8309-066fefb31240.png">

**Comment by alex-frankel** [comment]
Gotcha - I am seeing the same noise when I repro as well. Thank you for double-checking. We will investigate further.

**Comment by shenglol** [comment]
@alex-frankel Those are default values coming from the RP's Swagger, but they are in fact immutable properties. We need to ask the RP team to fix their Swagger.

**Comment by floriankoch** [comment]
The Noise is still present
```
 ~ microsoft.insights/components/xxxxxx [2020-02-02-preview]
    + properties.Flow_Type: "Bluefield"
```

**Comment by mikeblakeuk** [comment]
~ Microsoft.Insights/components/xxxxxx [2020-02-02]
    + properties.Flow_Type:      "Bluefield"
    + properties.Request_Source: "rest"

**Comment by torreymicrosoft** [comment]
Thanks for raising this, and for your patience. It was recently picked up as an outstanding open issue for the ARM Deployments team to review, and I am sorry it sat this long.

What-If for Azure Deployment Stacks is generally available as of today, and noise reduction is enabled in all regions. Your report is the textbook case for it: on `Microsoft.Insights/components`, `properties.Flow_Type: "Bluefield"` and `properties.Request_Source: "rest"` are values the Application Insights provider returns on its own. They were never in your template, which is why adding them by hand did not help either. Stacks What-If records a baseline when the stack is deployed and drops properties that are unchanged between that baseline and the current evaluation, so RP supplied values like these stop showing up as additions.

You were using PowerShell, so the equivalent is:

```powershell
New-AzResourceGroupDeploymentStackWhatIfResult `
  -Name appinsights-preview `
  -ResourceGroupName <your-rg> `
  -StackResourceId <your-stack-resource-id> `
  -TemplateFile main.bicep `
  -ActionOnUnmanage DetachAll `
  -DenySettingsMode None `
  -RetentionInterval PT3H
```

Two honest caveats, both from the docs. The baseline is recorded when the stack is deployed or updated, so a stack has to be deployed or updated once after the feature shipped before its results are filtered. And noise reduction removes many of these differences, not every one.

I am closing this one. If `Flow_Type` or `Request_Source` still appear through Stacks What-If, reopen this or file it in `Azure/deployment-stacks`. Thanks in particular for going back and re-running the deployment in a second subscription in 2020 to prove it was not the template, that is what made this report actionable.

Announcement: https://techcommunity.microsoft.com/blog/azuregovernanceandmanagementblog/now-generally-available-what-if-for-azure-deployment-stacks/4547614
Docs: https://learn.microsoft.com/azure/azure-resource-manager/bicep/deployment-stacks-what-if
